luigi_pipeline/sex_related_checks.py

```diff
@@ -0,0 +1,568 @@
+        # _SUCCESS file to make sure it was not terminated halfway.
+        return GCSorLocalTarget(f"{self.output().path}/_SUCCESS").exists()
+
+    def mt_path(self):
+        return GCSorLocalTarget(f"{self.output_directory().path}callset.mt")
+
+    def run(self):
+        # first validate paths
+        if self.hail_temp_dir:
+            hl.init(
+                tmp_dir=self.hail_temp_dir,
+            )  # Need to use the GCP bucket as temp storage for very large callset joins
+        hl._set_flags(
+            use_new_shuffle="1"
+        )  # Interval ref data join causes shuffle death, this prevents it
+
+        for source_path in [self.source_paths]:
+            check_if_path_exists(source_path, "source_path")
+        if self.wes_filter_source_paths:
+            for wes_filter_source_path in self.wes_filter_source_paths:
+                check_if_path_exists(source_path, "source_path")
+        if self.hail_temp_dir:
+            check_if_path_exists(self.hail_temp_dir, "hail_temp_dir")
+
+        self.read_input_write_mt()
+        self.call_sex()
+
+    def call_sex(
+        self,
+        use_y_cov: bool = False,
+        add_x_cov: bool = False,
+        y_cov_threshold: float = 0.1,
+        normalization_contig: str = "20",
+        xy_fstat_threshold: float = 0.75,
+        xx_fstat_threshold: float = 0.5,
+        aaf_threshold: float = 0.05,
+        call_rate_threshold: float = 0.25,
+        final_annotations: List[str] = [
+            "is_female",
+            "f_stat",
+            "n_called",
+            "expected_homs",
+            "observed_homs",
+            "sex",
+        ],
+    ):
+        """
+        Call sex for the samples in a given callset and export results file to the desired path.
+
+        :param MatrixTable mt: Input MatrixTable
+        :param use_y_cov: Set to True to calculate and use chrY coverage for sex inference. Default is False
+        :param add_x_cov: Set to True to calculate chrX coverage. Must be specified with use_y_cov. Default is False
+        :param y_cov_threshold: Y coverage threshold used to infer sex aneuploidies.
+        XY samples below and XX samples above this threshold will be inferred as having aneuploidies.
+        Default is 0.1
+        :param normalization_contig: Chosen chromosome for calculating normalized coverage. Default is "20"
+        :param xy_fstat_threshold: F-stat threshold above which a sample will be called XY. Default is 0.75
+        :param xx_fstat_threshold: F-stat threshold below which a sample will be called XX. Default is 0.5
+        :param aaf_threshold: Alternate allele frequency threshold for `hl.impute_sex`. Default is 0.05
+        :param call_rate_threshold: Minimum required call rate. Default is 0.25
+        """
+        # Read in matrix table and define output file name prefix
+
+        logger.info("Using chromosome Y coverage? %s", use_y_cov)
```

Comment:
I know we aren't doing this anywhere else, but it would be great if we could standardize our string formatting in this repo